packages/react-components/react-combobox/src/components/Combobox/useCombobox.tsx

```diff
@@ -18,22 +19,104 @@ import type { ComboboxProps, ComboboxState } from './Combobox.types';
  */
 export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLInputElement>): ComboboxState => {
   const baseState = useComboboxBaseState(props);
+  const {
+    activeOption,
+    clearSelection,
+    getIndexOfId,
+    getOptionsMatchingValue,
+    selectOption,
+    selectedOptions,
+    setActiveOption,
+    setOpen,
+    setValue,
+    value,
+  } = baseState;
+  const { allowFreeform, multiselect } = props;
 
   const { primary: triggerNativeProps, root: rootNativeProps } = getPartitionedNativeProps({
     props,
     primarySlotTagName: 'input',
     excludedPropNames: ['children', 'size'],
   });
 
+  // set active option and selection based on typing
+  const getOptionFromInput = (inputValue: string): OptionValue | null => {
+    // if there are commas in the value string, take the text after the last comma
+    const searchString = inputValue.split(',').pop();
+
+    if (!searchString || searchString.trim().length === 0) {
+      return null;
+    }
+
+    const matcher = (optionValue: string) => optionValue.toLowerCase().indexOf(searchString.trim().toLowerCase()) === 0;
```

Comment:
nit: instead of calling searchString.trim().toLowerCase() multiple times, you could call them once when it's initialized.

```suggestion
    const searchString = inputValue.split(',').pop()?.trim()?.toLowerCase();

    if (!searchString) {
      return null;
    }

    const matcher = (optionValue: string) => optionValue.toLowerCase().indexOf(searchString) === 0;
```